src/plugins/intel_cpu/src/nodes/kernels/scaled_attn/common.hpp

```diff
@@ -71,19 +88,44 @@ static constexpr size_t vec_len_f32_avx2 = vec_len_avx2 / sizeof(float);
 #endif
 
 #ifdef HAVE_AVX2
-    inline __m256 mm256_uni_loadu_ps(float* a) {
+    inline __m256i get_mask(int N7) {
+        static __m256i mask[] = {
+            _mm256_set_epi32(0, 0, 0, 0, 0, 0, 0, 0),
+            _mm256_set_epi32(0, 0, 0, 0, 0, 0, 0, -1),
+            _mm256_set_epi32(0, 0, 0, 0, 0, 0, -1, -1),
+            _mm256_set_epi32(0, 0, 0, 0, 0, -1, -1, -1),
+            _mm256_set_epi32(0, 0, 0, 0, -1, -1, -1, -1),
+            _mm256_set_epi32(0, 0, 0, -1, -1, -1, -1, -1),
+            _mm256_set_epi32(0, 0, -1, -1, -1, -1, -1, -1),
+            _mm256_set_epi32(0, -1, -1, -1, -1, -1, -1, -1),
+            _mm256_set_epi32(-1, -1, -1, -1, -1, -1, -1, -1),
+        };
+        return _mm256_loadu_si256(&mask[N7]);
+    }
+    inline __m256 mm256_uni_loadu_ps(const float* a) {
         return _mm256_loadu_ps(a);
     }
     inline void mm256_uni_storeu_ps(float* a,  __m256 v) {
         _mm256_storeu_ps(a, v);
     }
 
-    inline __m256 mm256_uni_loadu_ps(ov::bfloat16* a) {
-        auto vec_bf16 = _mm_loadu_si128(reinterpret_cast<__m128i*>(a));
+    inline __m256 mm256_uni_loadu_ps(const ov::bfloat16* a) {
+        auto vec_bf16 = _mm_loadu_si128(reinterpret_cast<const __m128i*>(a));
         auto o = _mm256_castsi256_ps(_mm256_slli_epi32(_mm256_cvtepu16_epi32(vec_bf16), 16));
         return o;
     }
 
+    inline __m256 mm256_uni_loadu_tail_ps(const float* a, const size_t count) {
+        auto mask = get_mask(count);
+        return _mm256_maskload_ps(a, mask);
+    }
+
+    inline __m256 mm256_uni_loadu_tail_ps(const ov::bfloat16* a, const size_t count) {
+        ov::bfloat16 tmp_values[8] = {0};
+        std::memcpy(tmp_values, a, count * sizeof(uint16_t));
+        return mm256_uni_loadu_ps(tmp_values);
+    }
```

Comment:
I saw this function is trying hard to support bf16 on AVX2 case, why do we need this?
bf16 infer-precision will be only used on AVX512 platform, right?

Reply:
OK, so you mean this overload is actually not supposed to be called in practice? if so, maybe we can just put an assert (with a comment or message) here, so:
- it can pass compilation
- no one will try to figure out why or waste time in trying harder to optimize it.